src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -246,6 +249,14 @@ private void PublishCode()
             _methodCodeNode.InitializeDebugVarInfos(_debugVarInfos);
         }
 
+#if READYTORUN
+        private void PublishEmptyCode()
+        {
+            _methodCodeNode.SetCode(new ObjectNode.ObjectData(Array.Empty<byte>(), null, 1, new ISymbolDefinitionNode[] { _methodCodeNode }));
```

Comment:
My concern with this would be about relocs to methods in this state - you go to great lengths to make these disappear from the various tables, but what if there's a direct call to a method like this (i.e. any place that could generate a reloc without the IsEmpty check)

Reply:
Hmm, one simple thing we could do is to avoid emitting the symbol in such case; that would make the cases you point out crash the compilation but that's better than emitting incorrect code. I still hope this problem might become less significant with your fix for the callsite intrinsics even though it's not likely to go away completely as I'm seeing other issues throwing exceptions during JIT compilation, for instance:

<pre>
    EXEC : warning : Method `[eh05]LocallocTest.func()` was not compiled because: [TEMPORARY EXCEPTION MESSAGE] InvalidProgramDefault: [D:\git\corert\tests_downloaded\CoreCLR\JIT\jit64\localloc\ehverify\eh07_dynamic\Test.csproj]
    EXEC : warning : Method `[Negative_ByRefLikeType]Test.ByRefLikeGenericInstantiation()` was not compiled because: [TEMPORARY EXCEPTION MESSAGE] ClassLoadGeneral: System.Collections.Generic.List`1, System.Runtime, Version=4.2.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a [D:\git\corert\tests_downloaded\CoreCLR\Loader\classloader\regressions\GitHub_11371\Negative_ByRefLikeType\Test.csproj]
    EXEC : warning : Method `[Negative_ByRefLikeType]Test.ByRefLikeFieldInNonByRefLikeType()` was not compiled because: [TEMPORARY EXCEPTION MESSAGE] ClassLoadGeneral: ByRefLikeFieldInNonValueType, Negative_ByRefLikeType, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null [D:\git\corert\tests_downloaded\CoreCLR\Loader\classloader\regressions\GitHub_11371\Negative_ByRefLikeType\Test.csproj]
    EXEC : warning : Method `[Negative_ByRefLikeType]Test.ByRefLikeFieldInNonValueType()` was not compiled because: [TEMPORARY EXCEPTION MESSAGE] ClassLoadGeneral: ByRefLikeFieldInNonValueType, Negative_ByRefLikeType, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null [D:\git\corert\tests_downloaded\CoreCLR\Loader\classloader\regressions\GitHub_11371\Negative_ByRefLikeType\Test.csproj]
    EXEC : warning : Method `[Negative_ByRefLikeType]Test.ByRefLikeStaticField()` was not compiled because: [TEMPORARY EXCEPTION MESSAGE] ClassLoadGeneral: ByRefLikeStaticField, Negative_ByRefLikeType, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null [D:\git\corert\tests_downloaded\CoreCLR\Loader\classloader\regressions\GitHub_11371\Negative_ByRefLikeType\Test.csproj]
    EXEC : warning : Method `[Negative_ByRefLikeType]Test.ByRefLikeBoxing()` was not compiled because: [TEMPORARY EXCEPTION MESSAGE] InvalidProgramSpecific: Void Test.ByRefLikeBoxing() [D:\git\corert\tests_downloaded\CoreCLR\Loader\classloader\regressions\GitHub_11371\Negative_ByRefLikeType\Test.csproj]
</pre>

Reply:
I don't think the callsite intrinsic fix is going to fix much more than the two tests. The `Negative_ByRefLikeType` tests look like a good problem to look at. How does crossgen deal with that? It's a method that references a type that is not loadable - that's why we can't compile it. How does crossgen deal with methods that call such uncompilable method? Where does the reloc point to?

(On CoreRT, we just go ahead and generate a throwing method body and call it a day.)